src/chain.h

```diff
@@ -475,7 +475,7 @@ class CChain {
 
     /** Return the maximal height in the chain. Is equal to chain.Tip() ? chain.Tip()->nHeight : -1. */
     int Height() const {
-        return vChain.size() - 1;
+        return (int)vChain.size() - 1;
```

Comment:
Why is this cast necessary?

Reply:
An unsigned integer overflow occurs in the case where `vChain.size() == 0`.

(And yes, the value return by `Height()` will be the expected `-1`, but an overflow occurred before returning :-))

Reply:
:+1: